frontends/concrete-python/tests/compilation/test_compiler.py

```diff
@@ -389,3 +389,44 @@ def f(x):
     with pytest.raises(ValueError) as excinfo:
         helpers.check_execution(circuit, f, sample)
     assert str(excinfo.value) == "Expected a valid type in dict_keys([])"
+
+
+def test_compiler_enable_fusing(helpers):
+    """
+    Test compilation with and without TLU fusing.
+    """
+
+    def f(x):
+        return (x**2) // 2
+
+    # Fused Scalar
+    compiler1 = Compiler(f, {"x": "encrypted"})
+    circuit1 = compiler1.compile(
+        fhe.inputset(fhe.uint3),
+        helpers.configuration().fork(enable_tlu_fusing=True),
+    )
+    assert circuit1.programmable_bootstrap_count == 1
+
+    # Not Fused Scalar
+    compiler2 = Compiler(f, {"x": "encrypted"})
+    circuit2 = compiler2.compile(
+        fhe.inputset(fhe.uint3),
+        helpers.configuration().fork(enable_tlu_fusing=False),
+    )
+    assert circuit2.programmable_bootstrap_count == 2
+
+    # Fused Tensor
+    compiler3 = Compiler(f, {"x": "encrypted"})
+    circuit3 = compiler3.compile(
+        fhe.inputset(fhe.tensor[fhe.uint3, 3]),  # type: ignore
+        helpers.configuration().fork(enable_tlu_fusing=True),
+    )
+    assert circuit3.programmable_bootstrap_count == 3
+
+    # Not Fused Tensor
+    compiler4 = Compiler(f, {"x": "encrypted"})
+    circuit4 = compiler4.compile(
+        fhe.inputset(fhe.tensor[fhe.uint3, 3]),  # type: ignore
+        helpers.configuration().fork(enable_tlu_fusing=False),
+    )
+    assert circuit4.programmable_bootstrap_count == 6
```

Comment:
- no need to check the default?
- no need to check print_tlu_fusing works as expected?
@BourgerieQuentin

Reply:
> no need to check print_tlu_fusing works as expected?

This is impossible to check as printing happens in C++.

> no need to check the default?

Default is just a boolean and as long as we test both True and False cases, we're testing the default.

Reply:
- OK for prints, but sad
- For default: I would add

```
  # Fused Tensor
    compiler3 = Compiler(f, {"x": "encrypted"})
    circuit3 = compiler3.compile(
        fhe.inputset(fhe.tensor[fhe.uint3, 3]),  # type: ignore
        helpers.configuration().fork(),
    )
    assert circuit3.programmable_bootstrap_count == 3
```

or something like that, to check we have really the default we want to have

Reply:
> helpers.configuration().fork()

This doesn't make sense as we're not testing with the default configuration, we have a custom test configuration.

> to check we have really the default we want to have

I'll include this piece of code in the beginning of this test:
```python
    # Make sure it's enabled by default
    default_configuration = fhe.Configuration()
    assert default_configuration.enable_tlu_fusing
```
as a commit to another PR of mine. Sounds good?

Reply:
yes

(I have not understood why my proposal doesn't make sense, you could take a default config and do what I said)